### ISSUE TITLE
Hourly entity does not report Hours

### ISSUE THREAD

**Issue by BerriJ** [hg1337/homeassistant-dwd#10]
Hi there,

thanks for this awesome integration! Today, I stumbled across my weather widget, which does not show the hours anymore. It just says "Friday", "Friday", "Friday" and so on...

![Bildschirmfoto vom 2023-08-04 13-13-56](https://github.com/hg1337/homeassistant-dwd/assets/37799358/d301ecb2-c53a-47e6-ae3e-966a719b0a98)

This may be somehow connected to the Update of Home Assistant 2023.08?

Best.
BerriJ

**Comment by hg1337** [comment]
Thank you BerriJ!

That's actually very strange. Recently, Home Assistant has indeed switched to a new system to provide the forecast. However, the previous method should still work. But there seems to be a new issue when more then 25 hours of hourly forecast is provided causing the effect you have in your screenshot. I have now additionally added the new way to provide the forecast which seems to solve this issue. I still needed to keep the old way in parallel to keep compatibility with 3rd party weather cards like https://github.com/bramkragten/weather-card. Seems to work so far. Please check the new version that I just released.

**Comment by BerriJ** [comment]
Thanks for your fast response. Your changes did fix the information view when clicking on the entity.

However, I made an additional observation. In Lovelace I'm using the home assistant weather card:
```
show_current: true
show_forecast: true
type: weather-forecast
name: Wuppertal
entity: weather.wuppertal_hourly
forecast_type: legacy
```
And this worked until the latest HA-Core Update. However, now it shows this:

![Bildschirmfoto vom 2023-08-05 09-33-07](https://github.com/hg1337/homeassistant-dwd/assets/37799358/3c1ab1ac-4ed8-4458-ac9c-8bbb1c790f0b)

Deleting `forecast_type: legacy` solves the issue. And the widget is displayed correctly. 

![Bildschirmfoto vom 2023-08-05 09-37-16](https://github.com/hg1337/homeassistant-dwd/assets/37799358/8df02394-90ed-41ad-a67d-9f08f8b1978f)

Just adding this for reference. No need to take further action. 

Thank you very much again.

**Comment by hg1337** [comment]
Thank you for the hint, I have added it to the release description.

**Comment by PhilippMundhenk** [comment]
Piggybacking on this thread, as I had the same issue. Update to newest version indeed solves the issue with the detailed forecast. However, I am having the same issue with the forecast not showing in the ```weather-forecast``` card. ```forecast_type: legacy``` is not set:

```
- type: weather-forecast
  entity: weather.wuppertal_hourly
  show_current: true
  show_forecast: true
```

@BerriJ : Did you happen to change anything else?

**Comment by hg1337** [comment]
@PhilippMundhenk If you are using the most recent version, you can also try the new weather.wuppertal entity (without suffix) with "forecast_type: hourly". You can also configure it in the Visual Editor.

**Comment by PhilippMundhenk** [comment]
Ah yes, that does it, thank you!

In the mean time, I also tried https://github.com/bramkragten/weather-card, which you referenced above and now I am not sure anymore, if I want to change back though ;)

**Comment by hg1337** [comment]
Glad that it works 🙂

I'm a bit worried that https://github.com/bramkragten/weather-card doesn't seem to be maintained any more. But it's still my favorite. Hope it still works for some time...
